doc/BACKUP.md

```diff
@@ -0,0 +1,312 @@
+* Create a RAID-1 `btrfs` filesystem.
+  * `mkfs.btrfs -m raid1 -d raid1 /dev/${HDD30GB} /dev/${USB32GB}`
+  * You may need to add `-f` if the USB flash disk is already formatted.
+* Create a mountpoint for the `btrfs` filesystem.
+* Create a `/etc/fstab` entry.
+  * Use the `UUID` option instad of `/dev/sdXX` since the exact device letter
+    can change across boots.
+  * You can get the UUID by `lsblk -o NAME,UUID`.
+    Specifying *either* of the devices is sufficient.
+  * e.g. `UUID=${UUID} ${BTRFSMOUNTPOINT} btrfs defaults 0 0`
+* `mount -a` then `df` to confirm it got mounted.
+* Copy the contents of the `$LIGHTNINGDIR` to the BTRFS mount point.
+  * Copy the entire directory, then `chown -R` the copy to the user who will
+    run the `lightningd`.
+  * If you are paranoid, run `diff -R` on both copies to check.
+* Remove the existing `$LIGHTNINGDIR`.
+* `ln -s ${BTRFSMOUNTPOINT}/lightningdirname ${LIGHTNINGDIR}`.
+  * Make sure the `$LIGHTNINGDIR` has the same structure as what you
+    originally had.
+* Add `crontab` entries for `root` that perform regular `btrfs` maintenance
+  tasks.
+  * `0 0 * * * /usr/bin/btrfs balance start -dusage=50 -dlimit=2 -musage=50 -mlimit=4 ${BTRFSMOUNTPOINT}`
+    This prevents BTRFS from running out of blocks even if it has unused
+    space *within* blocks, and is run at midnight everyday.
+    You may need to change the path to the `btrfs` binary.
+  * `0 0 * * 0 /usr/bin/btrfs scrub start -B -c 2 -n 4 ${BTRFSMOUNTPOINT}`
+    This detects bit rot (i.e. bad sectors) and auto-heals the filesystem,
+    and is run on Sundays at midnight.
+* Restart your Lightning node.
+
+If one or the other device fails completely, shut down your computer, boot
+on a recovery disk or similar, then:
+
+* Connect the surviving device.
+* Mount the partition/USB flash disk in `degraded` mode:
+  * `mount -o degraded /dev/sdXX /mnt/point`
+* Copy the `lightningd.sqlite3` and `hsm_secret` to new media.
+  * Do **not** write to the degraded `btrfs` mount!
+* Start up a `lightningd` using the `hsm_secret` and `lightningd.sqlite3`
+  and close all channels and move all funds to onchain cold storage you
+  control, then set up a new Lightning node.
+
+If the device that fails is the USB flash disk, you can replace it using
+BTRFS commands.
+You should probably stop your Lightning node while doing this.
+
+* `btrfs replace start /dev/sdOLD /dev/sdNEW ${BTRFSMOUNTPOINT}`.
+* Monitor status with `btrfs replace status ${BTRFSMOUNTPOINT}`.
+
+More sophisticated setups with more than two devices are possible.
+Take note that "RAID 1" in `btrfs` means "data is copied on up to two
+devices", meaning only up to one device can fail.
+You may be interested in `raid1c3` and `raid1c4` modes if you have
+three or four storage devices.
+BTRFS would probably work better if you were purchasing an entire set
+of new storage devices to set up a new node.
+
+On BSD you can use a ZFS RAID-Z setup, which is probably better than BTRFS
```

Comment:
The general advice (like first paragraph in this section also) could be moved together, BTRFS could be turned into its own sub-section so that it is optically skippable (sp?).

BTW this review comes from a point of view of a non-native speaker. So take it with a grain of salt.

Reply:
Okay.  I guess we can list software methods of RAID-1, such as:

* `mdadm`
* BTRFS
* ZFS